Lib/_pydatetime.py

```diff
@@ -545,7 +545,7 @@ def _isoweek_to_gregorian(year, week, day):
 def _check_tzname(name):
     if name is not None and not isinstance(name, str):
         raise TypeError("tzinfo.tzname() must return None or string, "
-                        "not '%s'" % type(name))
+                        "not '%s'" % type(name).__name__)
```

Comment:
If we're going for consistency, let's make this an f-string:
```suggestion
                        f"not {type(name).__name__!r}")
```